keps/sig-multicluster/5313-placement-decision-api/README.md

```diff
@@ -0,0 +1,537 @@
+// The referenced placement might not be a Kubernetes resource.
+// The name could be a correlation key understood by the scheduler and consumers.
+type PlacementRef struct {
+  // Name of the Placement that this decision answers.
+  // Must equal the value of the label multicluster.x-k8s.io/placement.
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=63
+  Name string `json:"name"`
+
+  // Optional: API group of the Placement resource if one exists.
+  // Omit when the placement is not represented as a Kubernetes object.
+  // +optional
+  APIGroup string `json:"apiGroup,omitempty"`
+
+  // Optional: kind of the Placement resource if one exists.
+  // Omit when the placement is not represented as a Kubernetes object.
+  // +optional
+  Kind string `json:"kind,omitempty"`
+
+  // Optional: namespace of the Placement resource when it is namespaced.
+  // Omit when the placement is not represented as a Kubernetes object or
+  // cluster scoped placement resource.
+  // +optional
+  Namespace string `json:"namespace,omitempty"`
+}
+```
+
+### API Example
+
+```
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: PlacementDecision
+metadata:
+  name: app-placement-decision-0
+  namespace: argocd
+  labels:
+    multicluster.x-k8s.io/decision-index: 0
+    multicluster.x-k8s.io/placement: app-placement
+schedulerName: multicluster-placement-controller
+placementRef:
+  apiGroup: multicluster.x-k8s.io
+  kind: Placement
+  namespace: argocd
+  name: app-placement
+decisions:
+- clusterProfileRef:
+    namespace: fleet1
+    name: cluster1
+  reason: "GPUs available"
+- clusterProfileRef:
+    namespace: fleet1
+    name: cluster2
+  reason: "GPUs available"
+```
+
+### Slicing
+
+* Following [EndpointSlice](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) design,
+  a single Placement can fan out to N `PlacementDecision` slices,
+  each limited to 100 clusters (`EndpointSlice`'s default).
+* All slices for one `Placement` MUST carry the same
+  `multicluster.x-k8s.io/placement=<placement-name>` label so consumers can List with a label selector.
+* If a scheduler needs to preserve the order of selected clusters and the result spans multiple slices,
+  it should label each PlacementDecision with `multicluster.x-k8s.io/decision-index=<index>`
+  where <index> starts at 0 and increments by 1.
+  Consumers that require ordering can sort by this label.
+
+### Lifecycle
+- **Create**: The scheduler creates the slice with the list of clusters in the decisions,
+  and set the label `multicluster.x-k8s.io/placement=<placement-name>`.
+  When order matters, set `multicluster.x-k8s.io/decision-index` on every slice.
+  The scheduler may choose to populate the reason for each decision for consumers/end-users
+  (ie, for debugging purposes).
+
+- **Update / Reschedule**: The scheduler may add or remove clusters in  decisions at any time.
```

Comment:
same comment as above, we need to make sure that the consumer knows when and if there is change in the decision.  I am not 100% sure that generation field works without spec.

Reply:
PlacementDecision changes should still be visible to consumers. Every update bumps the `resourceVersion` so standard informers fire an Update event each time the object changes and letting consumer controllers reconcile (or just diff the old and new decisions lists with a custom UpdateFunc in controller code). It should work the same way as how controllers handle `ConfigMap` with no spec/status updates.